src/ledger/LedgerTestUtils.cpp

```diff
@@ -70,20 +70,27 @@ makeValid(AccountEntry& a)
     }
 
     stripControlCharacters(a.homeDomain);
-    a.inflationDest.activate() = PubKeyUtils::random();
+
+    if (a.inflationDest)
+    {
+        *a.inflationDest = PubKeyUtils::random();
+    }
 
     std::sort(a.signers.begin(), a.signers.end(), &AccountFrame::signerCompare);
     a.signers.erase(
         std::unique(a.signers.begin(), a.signers.end(), signerEqual),
         a.signers.end());
     for (auto& s : a.signers)
     {
+        s.weight = s.weight & UINT8_MAX;
         if (s.weight == 0)
         {
             s.weight = 100;
         }
     }
     a.numSubEntries = (uint32)a.signers.size();
+    a.seqNum = a.seqNum & INT64_MAX;
```

Comment:
Does the C++ standard guarantee that it is well defined to do (uint64_t) & (int64_t)? There are several places in this PR where such operations occur.

Reply:
yes, it's just a bitwise operation

Reply:
But my understanding is that bitwise operations act on the representation of the data rather than the interpretation, and the C++ standard admits multiple representations of signed integers (although I am under the impression that almost all modern implementations are two's complement). However, I do not know if it is always guaranteed that a positive signed integer will have the same representation as an unsigned integer of the same number of bytes.

Reply:
positive numbers yes (they don't say "positive", rather "non-negative"):
per standard 3.9.1
 > The range of non-negative values of a signed integer type is a subrange of the corresponding unsigned integer type, and the value representation of each corresponding signed/unsigned type shall be the same.